replay2/memory_remapper/tests/memory_remapper_tests.cc

```diff
@@ -0,0 +1,205 @@
+// Copyright (C) 2022 Google Inc.
+//
+// This file is generated code. It was created by the AGI code generator.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include "replay2/memory_remapper/memory_remapper.h"
+
+using namespace agi::replay2;
+
+class ModResourceGenerator : public ResourceGenerator {
+public:
+	ModResourceGenerator(size_t length) : length_(length) {}
+	virtual ~ModResourceGenerator() {}
+
+	size_t length() const override { return length_; }
+
+	void generate(ReplayAddress replayAddress) override {
+		for(size_t i = 0; i < length_; ++i) {
+			replayAddress.bytePtr()[i] = std::byte(i % 256);
+		}
+	}
+private:
+	size_t length_;
+};
+
+class ConstResourceGenerator : public ResourceGenerator {
+public:
+	ConstResourceGenerator(std::byte value, size_t length) : value_(value), length_(length) {}
+	virtual ~ConstResourceGenerator() {}
+
+	size_t length() const override { return length_; }
+
+	void generate(ReplayAddress replayAddress) override {
+		for(size_t i = 0; i < length_; ++i) {
+			replayAddress.bytePtr()[i] = value_;
+		}
+	}
+private:
+	std::byte value_;
+	size_t length_;
+};
+
+void ASSERT_MOD_REPLAY_ADDRESS(const MemoryRemapper& remapper, const CaptureAddress& captureAddress, const ReplayAddress& replayAddress, size_t length) {
+	for(size_t i = 0; i < length; ++i) {
+		ReplayAddress replayAddress = remapper.RemapCaptureAddress(captureAddress.offsetByBytes(i));
```

Comment:
I think we should have functionality to avoid the loop here, with something like ReplayAddressRange replayRange = remapper.RemapCaptureRange...